### PR TITLE
Support external loadbalancers in EclCpGridVanguard

### PR DESCRIPTION
**Pull request by blattms**
Additionally adds a flow executable that only distributes cells in z-direction. Executable is named flow_distribute_z and uses the external load balancing information. It can be used to test the distributed standard wells on SPE9 with 4 or more processes.

Automatic test of this is still missing as I need to find how to best do this. Suggestions welcome.

Based on #3029  and hence needs OPM/opm-grid#511

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this opm-grid=511 please

**Comment by akva2** [comment]
A regular parallel test is certainly a good start. compares results from a serial and a paralllel simulation.

**Comment by blattms** [comment]
Thanks. Done.

**Comment by blattms** [comment]
jenkins build this opm-grid=511 please

File: opm/simulators/flow/BlackoilModelParametersEbos.hpp
Comment (totto82) [inline opm/simulators/flow/BlackoilModelParametersEbos.hpp:39]: Where is this used in eclbasevanguard.hh?

**Review by totto82 — APPROVED** [review]
Thanks. I have some small comments, (mostly for clarification) IMO this can be merged when data is provided to the new test case.

**Comment by akva2** [comment]
there is no data to provide, this is a parallel test not a regression test (ie it compares a sequential run to a parallel one).

**Comment by blattms** [comment]
Tets failed because forced push to opm-grid was missing.

File: opm/simulators/flow/BlackoilModelParametersEbos.hpp
Comment (blattms) [reply opm/simulators/flow/BlackoilModelParametersEbos.hpp:39]: Anoher remnant of the typetag fight. At the time I thought I could instantiate BlakoilModelParametersEbos in EclEclipseGrid. That or the thought that I had skills to do it turned out to be a quite naive assumption

**Review by blattms — COMMENTED** [review]
Will remove the remnants from my TypeTag approach

**Comment by blattms** [comment]
Typetag/property comments adressed. I needed one forward declaration in eclbasevanguard.hh

**Comment by blattms** [comment]
jenkins build this opm-grid=511 please

File: ebos/eclbasevanguard.hh
Comment (totto82) [reply ebos/eclbasevanguard.hh:245]: No problem. I am fine with your solution.

**Comment by totto82** [comment]
Thanks, from my side this is fine.

**Comment by blattms** [comment]
I thought I fixed the issue for the failing tests a while ago. Need to recheck.

**Comment by blattms** [comment]
Strange. According to the log [jenkins](https://ci.opm-project.org/job/opm-simulators-PR-builder/1780/consoleFull) still picks a previous commit [4c242d74](https://github.com/OPM/opm-grid/commit/4c242d74a33cec70333db07083d7983153331de2) instead of the force pushed on [200278080f](https://github.com/OPM/opm-grid/pull/511/commits/200278080fa8b409dc84b43a38272fe569622039) for opm-grid from OPM/opm-grid#511.

@akva2 Any ideas how to resolve this?

**Comment by akva2** [comment]
open/close pr cycle usually necessary to force github to update the merge refs.

**Comment by blattms** [comment]
Trying to jenkins to work properly

**Comment by blattms** [comment]
jenkins build this opm-grid=511 please

**Comment by blattms** [comment]
dance, jennkins,dance.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
shameless self-merge after approval.

**Pull request merged**
